### DISCUSSION TITLE
Proposal: Allow deconstructing a tuple of N items into fewer variables

### DISCUSSION THREAD

**Discussion by IanKemp** [dotnet/csharplang#5167] [General]
Given the method signature:

```
public (Foo foo, Bar bar) GimmehTuple() { ... }
```

we can deconstruct it like so:

```
var (foo, bar) = GimmehTuple();
```

But if the method's signature changes to include additional items, e.g:

```
public (Foo foo, Bar bar, Quux quux) GimmehTuple() { ... }
```

then the deconstruction code now generates a compiler error (CS8132 "Can't deconstruct a tuple of '3' elements into '2' variables.").

I propose that this should be allowed by having the compiler implicitly rewrite the given expression to insert discards. That is, the compiler-generated deconstruction code effectively becomes:

```
var (foo, bar, _) = GimmehTuple();
```

and compilation succeeds.

I'm sure there's an obvious reason why this wasn't implemented, I just can't see why it is. Anyone care to enlighten me?

**Comment by svick** [comment]
I don't know if this was *the* reason, but one possible reason is that if you declare too few variables by mistake, you want this to be an error. And since changing the return type this way is already a breaking change, requiring the extra `, _` shouldn't be a significant hurdle.

Also this discussion belongs on dotnet/csharplang, somebody will hopefully move it there.
